From 724e16aa8fbc060167beeb3faeb375537616fa2c Mon Sep 17 00:00:00 2001
From: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
Date: Fri, 2 Aug 2019 23:12:13 +0300
Subject: [PATCH] media: i2c: ov2311: put to autodetect head

This puts OV2311 in autodetect head becase the serializer
remote gpio is controlled from command line only

Imagers that have remote gpios in glue must be in tail

Signed-off-by: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
---
 drivers/media/i2c/soc_camera/ov106xx.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/media/i2c/soc_camera/ov106xx.c b/drivers/media/i2c/soc_camera/ov106xx.c
index 21dffcd8..eaa492e 100644
--- a/drivers/media/i2c/soc_camera/ov106xx.c
+++ b/drivers/media/i2c/soc_camera/ov106xx.c
@@ -91,6 +91,12 @@ static int ov106xx_probe(struct i2c_client *client,
 		goto out;
 	}
 
+	ret = ov2311_probe(client, did);
+	if (!ret) {
+		chip_id = ID_OV2311;
+		goto out;
+	}
+
 	ret = ov10635_probe(client, did);
 	if (!ret) {
 		chip_id = ID_OV10635;
@@ -145,12 +151,6 @@ static int ov106xx_probe(struct i2c_client *client,
 		goto out;
 	}
 
-	ret = ov2311_probe(client, did);
-	if (!ret) {
-		chip_id = ID_OV2311;
-		goto out;
-	}
-
 	ret = imx390_probe(client, did);
 	if (!ret) {
 		chip_id = ID_IMX390;
-- 
2.7.4

